src/client/auth/ticketZKProofMulti.ts

```diff
@@ -0,0 +1,148 @@
+import { AbstractAuthentication, AuthenticationMethodMulti, AuthenticationResult } from './abstractAuthentication'
+import { OffChainTokenConfig, MultiTokenInterface } from '../interface'
+import { OutletAction, Messaging } from '../messaging'
+import { Authenticator } from '@tokenscript/attestation'
+import { SignedUNChallenge } from './signedUNChallenge'
+import { UNInterface } from './util/UN'
+import { LocalOutlet } from '../../outlet/localOutlet'
+import { OutletIssuerInterface } from '../../outlet'
+import { createIssuerHashArray } from '../../utils'
+import { shouldUseRedirectMode } from '../../utils/support/getBrowserData'
+import { EasZkProof } from '@tokenscript/attestation/dist/eas/EasZkProof'
+import { DEFAULT_EAS_SCHEMA, TokenType } from '../../outlet/ticketStorage'
+import { EAS_RPC_CONFIG } from '../../core/eas'
+
+export class TicketZKProofMulti extends AbstractAuthentication implements AuthenticationMethodMulti {
+	TYPE = 'ticketZKProofMulti'
+
+	private messaging = new Messaging()
+
+	// NOTE: A limitation at this time is that the user can only authenticate tokens
+	// that use the same issuer origin, email and it's underlying config (see: firstCollectionFound via getTokenProofMulti).
```

Comment:
@nicktaras would be great to foreach tokens and if it use different origins or configs then throw new Error(). it will help to dont forget that limitation

Reply:
Thanks for this point. This is what has been done so far to address this, so at this point the tokens should only be from the same issuer origin (per batch request).

Via client getMultiRequestBatch() - we assign the tokens with the same tokenOrigin into batches.
authRequestBatch.offChain[issuerConfig.tokenOrigin]

As you mentioned we don't check the config settings, but think this should be enough for now to ensure they are using the correct issuer, letting the following process take hold of if the ticket is valid or not?

Where if I understand your thought on checking this config at this point, may help us to reject the multi-request earlier?

Reply:
I mean that we can forgot this limitation and get error
in case if we limit code according to some rule then we have to set simple check and in rule broken them throw an error with verbose description.
in cese if someone use incompable config then error will happen and we will know the reason. @nicktaras

Reply:
Miccy is working on the solution to this limitation at the moment, shared the PR for this. I'll close this thread @oleggrib